packages/next-codemod/bin/upgrade.ts

```diff
@@ -109,15 +115,21 @@ export async function runUpgrade(
     console.log(
       `${pc.green('✓')} Current Next.js version is already on the target version "v${targetNextVersion}".`
     )
-    endMessage()
-    return
+
+    if (!dry) {
+      endMessage()
+      return
+    }
```

Comment:
"dry" means by convention "no changes". The actual flow of the program shouldn't be altered. If you run `--dry` now, the program would tell you it does something that it won't which is confusing.

For testing, I'd recommend using the test fixtures. They automatically reset the install to before so you can just repeatedly test the fixture without having to revert to the original state.